### PR TITLE
Update migration generation and docs to use foreignKey()

### PR DESCRIPTION
**Pull request by markstory**
Use `foreignKey()` in migration generation (when `builtin` backend is active) and update the docs. The `foreignKey()` method was moved to `BaseMigration` to make code generation easier.

### CODE REVIEW

**Comment by LordSimal** [comment]
Love it 👍🏻 
But isn't this technically BC breaking since `foreignKey` doesn't exist on the Table class anymore?

File: src/Db/Table.php [hunk 1]
```diff
@@ -501,17 +501,6 @@ public function addForeignKey(string|array|ForeignKey $columns, string|TableValu
         return $this;
     }
 
-    /**
-     * Create a new ForeignKey object.
-     *
-     * @params string|string[] $columns Columns
-     * @return \Migrations\Db\Table\ForeignKey
```
Comment (dereuromark) [inline src/Db/Table.php:508]: would it be safer to deprecate only?

**Comment by LordSimal** [comment]
I just realized, that this method was not present in 4.x so this should be fine then 👍🏻

**Review by LordSimal — APPROVED** [review]

**Comment by markstory** [comment]
> But isn't this technically BC breaking since foreignKey doesn't exist on the Table class anymore?

I'm hoping to work around that in two ways by making this part of the opt-in on the new `builtin` backend. Bake will continue using the existing parameters until you enable the `builtin` backend. Once you've switched your base migration class to the new one, the migration level method (or table API) becomes available.

This could cause failures if someone were to switch, generate a new migration using the new API, revert the backend change. I don't know how to solve it or if it is an important scenario to cover. I would hope that developers wouldn't start using new APIs until they had completed testing out their upgrade.

**Comment by LordSimal** [comment]
> This could cause failures if someone were to switch, generate a new migration using the new API, revert the backend change.

I would not waste time trying to cover that usecase 😅 

I'd say it would be fine if we have proper documentation on what has changed in the builtin backend and what to be aware of when we inevitably change to the builtin backend by default

**Pull request merged**